File: 221018/theBiggestAndTheSmallest.test.js
import solution, {
  convertToArray, findTheBiggestNumber, findTheSmallestNumber, concatenate,
} from './theBiggestAndTheSmallest';

test('문자열을 숫자 배열로 만들기', () => {
  expect(convertToArray('-1 -2 -3 -4')).toStrictEqual([-1, -2, -3, -4]);
});

test('숫자 배열에서 최대값 추출', () => {
  expect(findTheBiggestNumber([-1, -2, -3, -4])).toBe(-1);
});

test('숫자 배열에서 최대값 추출 (같은 수인 경우)', () => {
  expect(findTheBiggestNumber([-1, -1, -1])).toBe(-1);
});

test('숫자 배열에서 최소값 추출', () => {
  expect(findTheSmallestNumber([-1, -2, -3, -4])).toBe(-4);
});

test('숫자 배열에서 최소값 추출 (같은 수인 경우)', () => {
  expect(findTheSmallestNumber([-1, -1, -1])).toBe(-1);
});

test('최대값과 최소값을 이어붙여서 문자열로 변환', () => {
  expect(concatenate(-1, -4)).toBe('-4 -1');
});

test('solution', () => {
  expect(solution('1 2 3 4')).toBe('1 4');
  expect(solution('-1 -2 -3 -4')).toBe('-4 -1');
  expect(solution('1 1 1 1')).toBe('1 1');
});

File: solutions/Previous/221017/TheBiggestAndTheSmallestTest.java
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class TheBiggestAndTheSmallestTest {
  @Test
  void slice() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    String numbers = "1 2 3 4";
    int[] slicedNumbers = new int[]{1, 2, 3, 4};

    assertThat(test.slice(numbers)).isEqualTo(slicedNumbers);
  }

  @Test
  void findTheBiggest() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    int[] numbers = new int[]{1, 2, 3, 4};
    int theBiggestNumber = 4;

    assertThat(test.findTheBiggest(numbers)).isEqualTo(theBiggestNumber);
  }

  @Test
  void findTheBiggestWhenEqual() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    int[] numbers = new int[]{1, 1};
    int theBiggestNumber = 1;

    assertThat(test.findTheBiggest(numbers)).isEqualTo(theBiggestNumber);
  }

  @Test
  void findTheSmallest() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    int[] numbers = new int[]{1, 2, 3, 4};
    int theSmallestNumber = 1;

    assertThat(test.findTheSmallest(numbers)).isEqualTo(theSmallestNumber);
  }

  @Test
  void findTheSmallestWhenEqual() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    int[] numbers = new int[]{1, 1};
    int theSmallestNumber = 1;

    assertThat(test.findTheSmallest(numbers)).isEqualTo(theSmallestNumber);
  }

  @Test
  void concatenate() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    int theSmallestNumber = 3;
    int theBiggestNumber = 10;
    String answer = "3 10";

    assertThat(test.concatenate(theSmallestNumber, theBiggestNumber))
        .isEqualTo(answer);
  }

  @Test
  void solution() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    String numbers = "1 2 3 4";
    String answer = "1 4";

    assertThat(test.solution(numbers)).isEqualTo(answer);
  }

  @Test
  void solutionWhenEqual() {
    TheBiggestAndTheSmallest test = new TheBiggestAndTheSmallest();

    String numbers = "-1 -1";
    String answer = "-1 -1";

    assertThat(test.solution(numbers)).isEqualTo(answer);
  }
}
